tests/tests/tx.rs

```diff
@@ -190,6 +198,142 @@ async fn dry_run_above_block_gas_limit() {
     }
 }
 
+fn arb_large_script_tx<R: Rng + rand::CryptoRng>(
+    max_fee_limit: Word,
+    size: usize,
+    rng: &mut R,
+) -> Transaction {
+    let mut script: Vec<_> = std::iter::repeat(op::noop()).take(size).collect();
+    script.push(op::ret(RegId::ONE));
+    let script_bytes = script.iter().flat_map(|op| op.to_bytes()).collect();
+    let mut builder = TransactionBuilder::script(script_bytes, vec![]);
+    let asset_id = *builder.get_params().base_asset_id();
+    builder
+        .max_fee_limit(max_fee_limit)
+        .script_gas_limit(22430)
+        .add_unsigned_coin_input(
+            SecretKey::random(rng),
+            rng.gen(),
+            u32::MAX as u64,
+            asset_id,
+            Default::default(),
+        )
+        .finalize()
+        .into()
+}
+
+fn config_with_size_limit(block_transaction_size_limit: u64) -> Config {
+    let mut chain_config = ChainConfig::default();
+    chain_config
+        .consensus_parameters
+        .set_block_transaction_size_limit(block_transaction_size_limit)
+        .expect("should be able to set the limit");
+    let state_config = StateConfig::default();
+
+    let mut config = Config::local_node_with_configs(chain_config, state_config);
+    config.block_production = Trigger::Never;
+    config
+}
+
+#[tokio::test]
+async fn transaction_selector_can_saturate_block_according_to_block_transaction_size_limit(
+) {
+    let mut rng = rand::rngs::StdRng::from_entropy();
+
+    // Create 5 transactions of increasing sizes.
+    let arb_tx_count = 5;
+    let transactions: Vec<(_, _)> = (0..arb_tx_count)
+        .map(|i| {
+            let script_bytes_count = 10_000 + (i * 100);
+            let tx =
+                arb_large_script_tx(189028 + i as Word, script_bytes_count, &mut rng);
+            let size = tx
+                .as_script()
+                .expect("script tx expected")
+                .metered_bytes_size() as u64;
+            (tx, size)
+        })
+        .collect();
+
+    // Run 5 cases. Each one will allow one more transaction to be included due to size.
+    for n in 1..=arb_tx_count {
+        // Calculate proper size limit for 'n' transactions
+        let block_transaction_size_limit: u64 =
+            transactions.iter().take(n).map(|(_, size)| size).sum();
+        let config = config_with_size_limit(block_transaction_size_limit);
+        let srv = FuelService::new_node(config).await.unwrap();
+        let client = FuelClient::from(srv.bound_address);
+
+        // Submit all transactions
+        for (tx, _) in &transactions {
+            let status = client.submit(tx).await;
+            assert!(status.is_ok())
+        }
+
+        let _ = tokio::time::sleep(Duration::from_millis(10)).await;
```

Comment:
Why do we need to wait? You can use the `srv.shared_state.txpool.insert` function if you want to be sure that the transaction is inserted. But  `client.submit(tx).await` should work as well

Reply:
It's a leftover from my initial test attempts. Removed in https://github.com/FuelLabs/fuel-core/pull/2195/commits/15a2cf7eaa1102cdfe93ed9eea8988a5b4278230.

cc: @MitchTurner - you may want to have a look at this.